dnf/conf/config.py

```diff
@@ -0,0 +1,929 @@
+            raise ValueError('invalid integer value')
+
+        if self._range_max is not None and n > self._range_max:
+            raise ValueError('Given value [%d] should be less than '
+                             'allowed value [%d].' % (n, self._range_max))
+        if self._range_min is not None and n < self._range_min:
+            raise ValueError('Given value [%d] should be greater than '
+                             'allowed value [%d].' % (n, self._range_min))
+        return n
+
+
+class PositiveIntOption(IntOption):
+    """An option representing a positive integer value, where 0 can
+    have a special representation.
+    """
+    def __init__(self, default=None, parent=None, runtimeonly=False,
+                 range_min=0, range_max=None, names_of_0=None):
+        self._names0 = [] if names_of_0 is None else names_of_0
+        super(PositiveIntOption, self).__init__(default, parent, runtimeonly,
+                                                range_min, range_max)
+
+    def _parse(self, s):
+        """Parse a string containing a positive integer, where 0 can
+           have a special representation."""
+        if s in self._names0:
+            return 0
+        return super(PositiveIntOption, self)._parse(s)
+
+
+class SecondsOption(Option):
+    """An option representing an integer value of seconds, or a human
+    readable variation specifying days, hours, minutes or seconds
+    until something happens. Works like :class:`BytesOption`.  Note
+    that due to historical president -1 means "never", so this accepts
+    that and allows the word never, too.
+
+    Valid inputs: 100, 1.5m, 90s, 1.2d, 1d, 0xF, 0.1, -1, never.
+    Invalid inputs: -10, -0.1, 45.6Z, 1d6h, 1day, 1y.
+
+    Return value will always be an integer
+    """
+    MULTS = {'d': 60 * 60 * 24, 'h' : 60 * 60, 'm' : 60, 's': 1}
+
+    def _parse(self, s):
+        """Parse a string containing an integer value of seconds, or a human
+        readable variation specifying days, hours, minutes or seconds
+        until something happens. Works like :class:`BytesOption`.  Note
+        that due to historical president -1 means "never", so this accepts
+        that and allows the word never, too.
+
+        Valid inputs: 100, 1.5m, 90s, 1.2d, 1d, 0xF, 0.1, -1, never.
+        Invalid inputs: -10, -0.1, 45.6Z, 1d6h, 1day, 1y.
+        """
+        if len(s) < 1:
+            raise ValueError("no value specified")
+
+        if s == "-1" or s == "never": # Special cache timeout, meaning never
+            return -1
+        if s[-1].isalpha():
+            n = s[:-1]
+            unit = s[-1].lower()
+            mult = self.MULTS.get(unit, None)
+            if not mult:
+                raise ValueError("unknown unit '%s'" % unit)
+        else:
+            n = s
+            mult = 1
+
+        try:
+            n = float(n)
+        except (ValueError, TypeError):
+            raise ValueError('invalid value')
```

Comment:
The value can be included and translation can improve it. Please can you make some changes?